### DISCUSSION TITLE
Analysis Engine - Ground Rules

### DISCUSSION THREAD

**Discussion by ansrlm** [githru/githru-vscode-ext#4] [General]
## !
거창한 rule 보다는 우리 모두가 간단하게 지킬 수 있으면서 유용한 협의사항들을 정리합니다.
많은 관심과 의견 부탁드립니다. (_ _)

### 목표
- 머지 기준에 대해 협의합니다.
  - 머지가 가능한 기준
    - 테스트 코드 통과 및 커버리지 통과
    - 승인 인원수
    - 자유
  - 머지를 하는 방식
    - merge
    - rebase
    - squash
    - 자유
- 리뷰 인원에 대해 협의합니다.
  - 모두를 reviewer로 지정
  - 특정 인원을 reviewer로 지정
    - 최소 인원이 필요할까요?
    - 특정 인원에게 몰리는 현상이 생길까요?
  - rand function을 통해서 reviewer로 지정
- 기술적 이슈 / 개인적 상황 등 공유가 필요한 경우 가용한 채널 방식에 대해 협의합니다.
  - 자유
  - 없음
  - 강제성이 없는 정기적 미팅 시간 (심심하면 들어오는 형태)
    - 온라인 + 오프라인
    - 온라인
- 이슈 / 위키 / 레이블에 대해 협의합니다.
  - 어느 상황에 이슈를 사용하나요?
  - 어느 상황에 위키를 사용하나요?
  - 각 상황에 맞는 레이블들은 무엇이 있을까요?
    - @analysis-engine
    - priority: ${something}
    - ...

**Comment by hy57in** [comment]
### 머지가 가능한 기준 
  - 테스트 코드 통과
  - 승인 인원수 : 2명
### 머지를 하는 방식
  -   squash (이유: 커밋 기록이 깔끔해지고 관리가 편함. revert 하기도 편함)
### 리뷰 인원
  - 모두를 reviewer로 지정
### 기술적 이슈 / 개인적 상황 등 공유가 필요한 경우 가용한 채널 방식
  - 1주일에 한번 미팅 오프라인 대관! 가능한 사람은 오프라인 참석, 온라인으로 참석도 가능
  - 수시로 모각코 열기! 나 지금 작업하고 있는데 같이 할 사람~
### 이슈 / 위키 / 레이블
  - 이슈 : PR이 생성될 만한 것
  - 위키 : 문서를 정리해야할 때
  - 라벨 : `analysis engine` 필수로 달기, 우선순위 라벨은 선택

**Reply by hy57in** [reply]
label도 자동으로 달아주는 action을 만들 수도 ..
컨트리뷰터봇도 달 수도..

**Comment by ansrlm** [comment]
### 협의 사항
- 온라인 / 오프라인 미팅은 사정에 맞게 강제성 없이 자율
- 이전 회의에서 다음 회의에 대한 협의 진행
- 금주 피쳐 리스트 협의를 위한 회의 (화 21:00) 진행
- 금주 모각코 및 진행방향 공유를 위한 회의 (토 13:00) 진행 및 OSS 화요일에 문의 (문기)
